File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.cpp
class Solution {
public:
    int maxDepth(string s) 
    {
        int maxDepth=0;
        vector<int> stack;
        for(char c : s)
        {
            if(c=='(')
            stack.push_back(c);
            else if(c==')')
            stack.pop_back();

            maxDepth=max((int)stack.size(),maxDepth);
        }
        return maxDepth;
    }
};

File: 1737-maximum-nesting-depth-of-the-parentheses/maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int maxDepth = 0, depth = 0;
        for(char ch : s.toCharArray()){
            if(ch == '('){
                depth++;
            }else if(ch == ')'){
                depth--;
            }
            maxDepth = Math.max(maxDepth, depth);
        }
        return maxDepth;
    }
}